server/kafka/mock_schema_registry.go

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2019-2022 The NATS Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package kafka
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/riferrei/srclient"
+	"github.com/stretchr/testify/assert"
+)
+
+// schema(s) for testing
+const avroSchema = "{" +
+	"\"type\": \"record\"," +
+	"\"name\": \"snack\"," +
+	"\"fields\": [" +
+	"{\"name\": \"name\", \"type\": \"string\"}," +
+	"{\"name\": \"manufacturer\", \"type\": \"string\"}," +
+	"{\"name\": \"calories\", \"type\": \"float\"}," +
+	"{\"name\": \"color\", \"type\": [\"null\", \"string\"], \"default\": null}" +
+	"]}"
+
+// sample message(s) for testing
+const avroMessage = "{\"name\":\"cookie\", \"manufacturer\": \"cadbury\", \"calories\": 100.0}"
+
+// schema registry subject name(s)
+const avroSubjectName = "snack_avro-value"
+
+// schema version(s)
+const avroSchemaVersion = 1
+const avroSchemaVersionString = "1"
+
+// schema id(s)
+const avroSchemaID = 1
+const avroSchemaIDString = "1"
+
+type schemaResponse struct {
+	Subject    string               `json:"subject"`
+	Version    int                  `json:"version"`
+	Schema     string               `json:"schema"`
+	SchemaType *srclient.SchemaType `json:"schemaType"`
+	ID         int                  `json:"id"`
+	References []srclient.Reference `json:"references"`
+}
+
+type MockSchemaRegistry struct {
+	instance *httptest.Server
+	t        *testing.T
+}
+
+// NewMockSchemaServer Mock schema registry instance which is used to test the producer and consumer functionality
+func NewMockSchemaServer(t *testing.T) MockSchemaRegistry {
```

Comment:
Can we unexport `MockSchemaRegistry` and `NewMockSchemaServer`? Let's keep them private first. If needed, we can make them public later.

Reply:
Done.